File: examples/Erlang/hwserver.erl
-module(hwserver).
-export([main/0]).

%% Starts a local hello server.

main() ->
    application:start(chumak),
    {ok, Socket} = chumak:socket(rep, "my-rep"),
    {ok, Pid} = chumak:bind(Socket, tcp, "localhost", 5555),
    loop(Socket).

loop(Socket) ->
    {ok, RecvMessage} = chumak:recv(Socket),
    io:format("Received request : ~p\n", [RecvMessage]),

    timer:sleep(1000),

    chumak:send(Socket, "World"),
    loop(Socket).

File: examples/Rust/hwserver.rs
#![crate_name = "hwserver"]

extern crate zmq;

use std::{thread, time};

fn main() {
    let context = zmq::Context::new();
    let responder = context.socket(zmq::REP).unwrap();
    assert!(responder.bind("tcp://*:5555").is_ok());

    loop {
        let buffer = &mut [0; 10];
        responder.recv_into(buffer, 0).unwrap();
        println!("Received Hello");
        thread::sleep(time::Duration::from_secs(1));
        responder.send_str("World", 0).unwrap();
    }
}
